src/components/InlineQuickReplyList/InlineQuickReplyList.tsx

```diff
@@ -0,0 +1,157 @@
+  unknown,
+  TockTheme
+> = styled.div`
+  display: flex;
+  overflow: auto;
+  -webkit-overflow-scrolling: touch;
+  justify-content: start;
+  scroll-behavior: smooth;
+  touch-action: pan-x pan-y;
+  align-items: center;
+  position: relative;
+  &::-webkit-scrollbar {
+    display: none;
+  }
+  scrollbar-width: none;
+  margin: 0.5em auto;
+
+  & > * {
+    flex-shrink: 0;
+  }
+`;
+
+const Previous: StyledComponent<
+  DetailedHTMLProps<HTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
+  unknown,
+  TockTheme
+> = styled.button`
+  position: absolute;
+  margin: auto;
+  left: 0;
+  top: 0;
+  bottom: 0;
+  padding: 1em;
+  background: ${(props) => opacify(-0.8, props.theme.palette.background.bot)};
+  border: none;
+  width: 3em;
+  height: 3em;
+  border-radius: 50%;
+
+  cursor: pointer;
+  z-index: 5;
+
+  & svg {
+    stroke: ${prop<any>('theme.palette.background.bot')};
+  }
+
+  &:hover,
+  &:focus {
+    svg {
+      stroke: ${prop<any>('theme.palette.text.bot')};
+    }
+  }
+
+  ${prop<any>('theme.overrides.quickReplyArrow', '')};
+`;
+
+const Next: StyledComponent<
+  DetailedHTMLProps<HTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
+  unknown,
+  TockTheme
+> = styled.button`
+  position: absolute;
+  margin: auto;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  padding: 1em;
+  background: ${(props) => opacify(-0.8, props.theme.palette.background.bot)};
+  border: none;
+  width: 3em;
+  height: 3em;
+  border-radius: 50%;
+
+  cursor: pointer;
+  z-index: 5;
+
+  & svg {
+    stroke: ${prop<any>('theme.palette.background.bot')};
+  }
+
+  &:hover,
+  &:focus {
+    svg {
+      stroke: ${prop<any>('theme.palette.text.bot')};
+    }
+  }
+
+  ${prop<any>('theme.overrides.quickReplyArrow', '')};
+`;
+
+const InlineQuickReplyListOuterContainer: StyledComponent<
+  DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
+  unknown,
+  TockTheme
+> = styled.div`
+  position: relative;
+`;
+type Props = {
+  items: Button[];
+  onItemClick: (button: Button) => void;
+};
+
+const InlineQuickReplyList = ({ items, onItemClick }: Props) => {
+  const theme: TockTheme = useTheme<TockTheme>();
+  const [ref, previous, next] = useCarouselQuickReply<HTMLDivElement>(
+    items?.length,
+  );
+
+  const [leftVisible, rightVisible] = useArrowVisibility(
+    ref.container,
+    ref.items,
+  );
+
+  return (
+    <InlineQuickReplyListOuterContainer>
+      {leftVisible && (
+        <Previous onClick={previous}>
+          <ArrowLeftCircle size={`${theme.typography.fontSize}`} />
+        </Previous>
+      )}
+      <InlineQuickReplyListContainer ref={ref.container}>
+        {items.map((child, index) =>
+          React.cloneElement(
+            <QuickReply
+              key={`${child.label}-${index}`}
+              onClick={onItemClick.bind(null, child)}
+              {...child}
+            />,
+            { ref: ref.items[index] },
+            undefined,
+          ),
```

Comment:
Given that parents cannot get direct refs to the `QuickReply` here, couldn't this be simplified to
```suggestion
            <QuickReply
              key={`${child.label}-${index}`}
              onClick={onItemClick.bind(null, child)}
              {...child}
              ref={ref.items[index]}
            />
```

Reply:
I already tried this implementation bud it did not work, I do not know really why unless I did a mistake somewhere else.

Reply:
I did a mistake last time for sure, it seems to work. I merged your implementation.